api/v1alpha1/warehouse_types.go

```diff
@@ -267,6 +284,127 @@ type WarehouseStatus struct {
 	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,4,opt,name=observedGeneration"`
 	// LastFreight refers to the last Freight produced by this Warehouse
 	LastFreight *FreightReference `json:"lastFreight,omitempty" protobuf:"bytes,5,opt,name=lastFreight"`
+	// DiscoveredArtifacts holds the artifacts discovered by the Warehouse.
+	DiscoveredArtifacts *DiscoveredArtifacts `json:"discoveredArtifacts,omitempty" protobuf:"bytes,7,opt,name=discoveredArtifacts"`
+}
+
+// DiscoveredArtifacts holds the artifacts discovered by the Warehouse.
+type DiscoveredArtifacts struct {
+	// Commits holds the Git commits discovered by the Warehouse.
+	Commits []GitDiscoveryResult `json:"commits,omitempty" protobuf:"bytes,1,rep,name=commits"`
+
+	// Images holds the images discovered by the Warehouse.
+	Images []ImageDiscoveryResult `json:"images,omitempty" protobuf:"bytes,2,rep,name=images"`
+
+	// Charts holds the charts discovered by the Warehouse.
+	Charts []ChartDiscoveryResult `json:"charts,omitempty" protobuf:"bytes,3,rep,name=charts"`
+}
+
+// GitDiscoveryResult represents the result of a Git discovery operation for a
+// GitSubscription.
+type GitDiscoveryResult struct {
+	// RepoURL is the repository URL of the GitSubscription.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern=`^https?://(\w+([\.-]\w+)*@)?\w+([\.-]\w+)*(:[\d]+)?(/.*)?$`
+	RepoURL string `json:"repoURL" protobuf:"bytes,1,opt,name=repoURL"`
+
+	// Commits is a list of commits discovered by the Warehouse for the
+	// GitSubscription. An empty list indicates that the discovery operation was
+	// successful, but no commits matching the GitSubscription criteria were found.
+	// +optional
+	Commits []DiscoveredCommit `json:"commits,omitempy" protobuf:"bytes,2,rep,name=commits"`
+}
+
+// DiscoveredCommit represents a commit discovered by a Warehouse for a
+// GitSubscription.
+type DiscoveredCommit struct {
+	// ID is the identifier of the commit. This typically is a SHA-1 hash.
```

Comment:
"typically?" Is it ever not? (Sincere question.)

Reply:
While it has been a multi-year-long process, Git is still paving the way for SHA-256. As a matter of fact, since Git 2.42 it is no longer marked as experimental.

Reply:
TIL